pkg/apis/networking/validation/validation.go

```diff
@@ -209,10 +209,10 @@ func validateIngressTLS(spec *networking.IngressSpec, fldPath *field.Path) field
 func ValidateIngressSpec(spec *networking.IngressSpec, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	// TODO: Is a default backend mandatory?
-	if spec.Backend != nil {
-		allErrs = append(allErrs, validateIngressBackend(spec.Backend, fldPath.Child("backend"))...)
+	if spec.DefaultBackend != nil {
+		allErrs = append(allErrs, validateIngressBackend(spec.DefaultBackend, fldPath.Child("defaultBackend"))...)
```

Comment:
this will produce confusing validation messages for existing v1beta1 API callers... since the `defaultBackend` field doesn't exist in their API versions. for this to be coherent, this needs to know either what version is being validated, or the field name to use for the backend field

Reply:
This is a fundamental problem with validation today - it always runs on the INTERNAL type, which means error messages in this situation are confusing.

I don't know of a better solution.  https://github.com/kubernetes/kubernetes/issues/8550

Suggestions welcome.

Reply:
I spoke to @lavalamp about versioned validation.  He said he'd support an approach to start moving to versioned validation, but wasn't sure exactly what the right hook is.  If we want to start exploring that in parallel, I'm happy to try to help try to figure it out.

In the mean time, I think we should go with the new name in errors.

The alternative is to plumb operated-as-version strings down into validation (and updates).  Blech.

Reply:
> In the mean time, I think we should go with the new name in errors.
> 
> The alternative is to plumb operated-as-version strings down into validation (and updates). Blech.

as a user, it is maddening to get an API error, follow the instructions, and not resolve the error (or produce a different "unknown field" error). I'd probably plumb a map of internal to external name for this one until the per-version validation approach materializes

Reply:
So should we switch the err to include the old spec field in there or do we want to go more complex?

Reply:
I don't think the plumbing to get the original-operation-version passed all the way down to validation is going to fly.  It's almost as ugly as hacking in versioned validation.  This info is lost deep deep in the bowels of the conversion logic.

I see only bad options here:

1) don't rename
2) return errors with the wrong name in some cases (one way or the other)
3) return errors with both names
4) block this on getting real plumbing laid

Every time I explain that `backend` is the default backend, people ask me why it's not called `default Backend`.  I lean towards (2) or (3).

What if we said "defaultBackend (aka backend in v1beta1)" here and left a big comment blaming @lavalamp for not giving us versioned validation?